c++/src/capnp/compiler/generics.c++

```diff
@@ -574,7 +582,15 @@ kj::Maybe<BrandedDecl> BrandScope::compileDeclExpression(
     case Expression::ABSOLUTE_NAME: {
       auto name = source.getAbsoluteName();
       KJ_IF_MAYBE(r, resolver.getTopScope().resolver->resolveMember(name.getValue())) {
-        return interpretResolve(resolver, *r, source);
+        auto result = interpretResolve(resolver, *r, source);
+        if (r->is<Resolver::ResolvedDecl>()){
+          errorReporter.reportResolution(Resolution {
```

Comment:
I'm not confident since it's not handling recursion.

Reply:
@kentonv 
Could you review it again when you have time?

Reply:
This will correctly handle the `Foo` in `foo @0 :.Foo`, and `RELATIVE_NAME` will correctly handle `Bar` in `bar @1 :Bar`.

But if you have `baz :2 :Baz.Qux.Corge`, I think `Qux` and `Corge` aren't currently handled. To handle them you need to add code to the `Expression::MEMBER` case, below.

Reply:
@kentonv 

That makes sense. My plugin is working as you described.
Could you tell me how to extract Type ID from `memberDecl`?
I don't know the way to access to the `ResolvedDecl` of `MEMBER`. Maybe I need to call `asResolveResult()`.

Reply:
Hmm, looks like we should add a method `BrandedDecl::getGenericTypeId()`, with the documentation:

> If this declaration points to a type, gets its type ID. Keep in mind that this drops the brand, that is, if the type is the result of applying a generic to some type parameters, this returns only the ID of the underlying generic type, which loses information about the type parameters.

That said, it occurs to me that we've designed this in a way where jump-to-declaration will skip aliases. That is, if the symbol points to an alias defined with `using`, we will jump straight through it and to the thing that alias points at. That's probably OK most of the time, and doing better would require deeper refactoring I think. So, let's just stick with it, but I wanted to point it out as an issue.

Reply:
I added the `kj::Maybe<uint64_t> BrandedDecl::getGenericTypeId()` with the documentation.
Thank you for pointing out the behavior regarding aliases.